cardano-node/src/Cardano/Node/Configuration/POM.hs

```diff
@@ -531,6 +518,20 @@ makeNodeConfiguration pnc = do
                  EnabledP2PMode  -> SomeNetworkP2PMode Consensus.EnabledP2PMode
                  DisabledP2PMode -> SomeNetworkP2PMode Consensus.DisabledP2PMode
              }
+ where
+   makeShutdownConfig :: PartialShutdownConfig -> Either String ShutdownConfig
```

Comment:
This resulted from conflicting changes of the shutdown machinery and tracers done by you and Serge. I'm currently unable to cope with this.

Reply:
> I'm currently unable to cope with this.
I don't understand. What do you mean?

Reply:
@Jimbo4350, while integrating the new tracing system, we have faced circular dependencies -- essentially coming from `Cardano.Node.Types` carrying too many things together.

So, we had to increase modularity -- by splitting off some of the components -- which meant they now have to define their own configuration (instead of relying on the single central `NodeConfiguration`).

As a consequence, configuration subtypes (`ShutdownConfig` and `SocketConfig`) were born.

Since this is the first ever split of configuration subtypes, it indeed has no precedent in the codebase.

Reply:
No worries. If the required changes aren't clear, I'm happy to make them. Let me know when is ok to do so.

Reply:
Sure!

Reply:
..and thank you for agreeing to handle it!